File: Python/lambdaconscarcdr.py
#!/usr/bin/env python

kons  = lambda hd, tl=None: lambda x: hd if x else tl
virst = lambda l: l(True )
rrest = lambda l: l(False)

lstt  = kons(1, kons(2))

virst(rrest(lstt))

File: PHP/lambdaconscarcdr.php
#!/usr/bin/env php

<?php
$kons  = function ($hd, $tl = null) {
    return function ($x) use ($hd, $tl) {
        return $x ? $hd : $tl;
    };};

$virst = function ($l) { return $l(true ); };
$rrest = function ($l) { return $l(false); };

$lstt  = $kons(1, $kons(2));

$virst($rrest($lstt));
?>
